python/ray/data/impl/compute.py

```diff
@@ -259,4 +272,11 @@ def get_compute(compute_spec: Union[str, ComputeStrategy]) -> ComputeStrategy:
     elif isinstance(compute_spec, ComputeStrategy):
         return compute_spec
     else:
-        raise ValueError("compute must be one of [`tasks`, `actors`]")
+        if isinstance(compute_spec, str):
+            match = re.match(ACTOR_POOL_REGEX, compute_spec)
```

Comment:
What do you think about decoupling the actor pool configuration, `ActorPoolSpec`, from the actor pool class, `ActorPool`, and expose `ActorPoolSpec` in the API instead of regexing on a string? The former is easier to extend and document, and I think that the latter is generally considered an API design anti-pattern.